apps/system/utils/utils_hwconfig.c

```diff
@@ -0,0 +1,99 @@
+/****************************************************************************
+ * apps/system/utils/utils_hwconfig.c
+ *
+ *   Copyright (C) 2013 Gregory Nutt. All rights reserved.
+ *   Author: Gregory Nutt <gnutt@nuttx.org>
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name NuttX nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <errno.h>
+#include <stdio.h>
+#include <fcntl.h>
+
+#include <tinyara/hwconfig.h>
+#include <tinyara/fs/ioctl.h>
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name : utils_hwconfig
+ *
+ * Description:
+ *   It reads the hw configurations from hwconfig drivers
+ *   and prints the status like swd enable/disable.
+ *
+ ****************************************************************************/
+int utils_hwconfig(int argc, char **args)
+{
+	int hwconfig_fd;
+	int ret;
+	hwconfig_fd = open(HWCONFIG_DRVPATH, O_RDWR);
+	if (hwconfig_fd < 0) {
+		printf("Fail to open hwconfig, errno %d.\n", get_errno());
+		return ERROR;
+	}
+
+	ret = ioctl(hwconfig_fd, HWCONFIGIOC_SWD, 0);
```

Comment:
How about using `read` instead of `ioctl`? Because it has only read operation.

Reply:
how about this?

Reply:
We can read one of `SWD` or `DOWNLOAD` separately.
So I think ioctl is better.